integration-tests/pom.xml

```diff
@@ -1,110 +1,103 @@
-            </plugin>
-            <plugin>
-                <artifactId>maven-failsafe-plugin</artifactId>
-                <executions>
-                    <execution>
-                        <goals>
-                            <goal>integration-test</goal>
-                            <goal>verify</goal>
-                        </goals>
-                        <configuration>
-                            <systemPropertyVariables>
-                                <native.image.path>${project.build.directory}/${project.build.finalName}-runner
-                                </native.image.path>
-                                <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
-                                <maven.home>${maven.home}</maven.home>
-                            </systemPropertyVariables>
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>
-        </plugins>
-    </build>
-    <profiles>
-        <profile>
-            <id>native-image</id>
-            <activation>
-                <property>
-                    <name>native</name>
-                </property>
-            </activation>
-            <build>
-                <plugins>
-                    <plugin>
-                        <artifactId>maven-surefire-plugin</artifactId>
-                        <configuration>
-                            <skipTests>${native.surefire.skip}</skipTests>
-                        </configuration>
-                    </plugin>
-                </plugins>
-            </build>
-            <properties>
-                <skipITs>false</skipITs>
-                <quarkus.package.type>native</quarkus.package.type>
-            </properties>
-        </profile>
-    </profiles>
-</project>
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd"
+         xmlns="http://maven.apache.org/POM/4.0.0">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>io.quarkiverse.wiremock</groupId>
+        <artifactId>quarkus-wiremock-parent</artifactId>
+        <version>0.0.3-SNAPSHOT</version>
+    </parent>
+    <artifactId>quarkus-wiremock-integration-tests</artifactId>
+    <name>WireMock - Integration Tests</name>
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-reactive</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-rest-client-reactive</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkiverse.wiremock</groupId>
+            <artifactId>quarkus-wiremock</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>

```

Comment:
If you remove the provided scope the `QuarkusIntegrationTest` will also work without any warnings.

Before:
```
[INFO] Running io.quarkiverse.wiremock.devservices.WireMockDevServiceResourceIT
2023-09-26 11:38:33,512 DEBUG [io.qua.wir.dev.WireMockDevProcessor] (build-1) Quarkus launch mode [TEST]
2023-09-26 11:38:33,513 DEBUG [io.qua.wir.dev.WireMockDevProcessor] (build-1) Starting WireMock server with port [8089] and path [src/test/resources]
2023-09-26 11:38:33,655 INFO  [org.ecl.jet.uti.log] (build-1) Logging initialized @1303ms to org.eclipse.jetty.util.log.Slf4jLog
2023-09-26 11:38:33,783 INFO  [org.ecl.jet.ser.Server] (build-1) jetty-9.4.49.v20220914; built: 2022-09-14T01:07:36.601Z; git: 4231a3b2e4cb8548a412a789936d640a97b1aa0a; jvm 11.0.19+7
2023-09-26 11:38:33,794 INFO  [org.ecl.jet.ser.han.ContextHandler] (build-1) Started o.e.j.s.ServletContextHandler@2a6718f9{/__admin,null,AVAILABLE}
2023-09-26 11:38:33,801 INFO  [org.ecl.jet.ser.han.Con.ROOT] (build-1) RequestHandlerClass from context returned com.github.tomakehurst.wiremock.http.StubRequestHandler. Normalized mapped under returned 'null'
2023-09-26 11:38:33,801 INFO  [org.ecl.jet.ser.han.ContextHandler] (build-1) Started o.e.j.s.ServletContextHandler@6a6b8fd9{/,null,AVAILABLE}
2023-09-26 11:38:33,814 INFO  [org.ecl.jet.ser.AbstractConnector] (build-1) Started NetworkTrafficServerConnector@7c4dc9ee{HTTP/1.1, (http/1.1, h2c)}{0.0.0.0:8089}
2023-09-26 11:38:33,814 INFO  [org.ecl.jet.ser.Server] (build-1) Started @1463ms
2023-09-26 11:38:33,945 INFO  [org.ecl.jet.ser.han.Con.__admin] (qtp988688045-27) RequestHandlerClass from context returned com.github.tomakehurst.wiremock.http.AdminRequestHandler. Normalized mapped under returned 'null'
Executing "/Users/wjglerum/.sdkman/candidates/java/11.0.19-tem/bin/java -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/Users/wjglerum/Documents/Dev/quarkus-wiremock/integration-tests/target/quarkus.log -Dquarkus.log.file.enable=true -Dquarkus.wiremock.devservices.port=8089 -jar /Users/wjglerum/Documents/Dev/quarkus-wiremock/integration-tests/target/quarkus-app/quarkus-run.jar"
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2023-09-26 11:38:34,409 WARN  [io.qua.config] (main) Unrecognized configuration key "quarkus.wiremock.devservices.port" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
2023-09-26 11:38:34,661 INFO  [io.quarkus] (main) quarkus-wiremock-integration-tests 0.0.3-SNAPSHOT on JVM (powered by Quarkus 2.16.5.Final) started in 0.418s. Listening on: http://0.0.0.0:8081
2023-09-26 11:38:34,661 INFO  [io.quarkus] (main) Profile prod activated. 
2023-09-26 11:38:34,661 INFO  [io.quarkus] (main) Installed features: [cdi, rest-client-reactive, resteasy-reactive, smallrye-context-propagation, vertx]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.813 s -- in io.quarkiverse.wiremock.devservices.WireMockDevServiceResourceIT
```
Here the `wiremock` extension doesn't end up in the produced jar

After removing the `provided` scope:
```
[INFO] Running io.quarkiverse.wiremock.devservices.WireMockDevServiceResourceIT
2023-09-26 11:37:41,725 DEBUG [io.qua.wir.dev.WireMockDevProcessor] (build-2) Quarkus launch mode [TEST]
2023-09-26 11:37:41,726 DEBUG [io.qua.wir.dev.WireMockDevProcessor] (build-2) Starting WireMock server with port [8089] and path [src/test/resources]
2023-09-26 11:37:41,875 INFO  [org.ecl.jet.uti.log] (build-2) Logging initialized @1378ms to org.eclipse.jetty.util.log.Slf4jLog
2023-09-26 11:37:42,023 INFO  [org.ecl.jet.ser.Server] (build-2) jetty-9.4.49.v20220914; built: 2022-09-14T01:07:36.601Z; git: 4231a3b2e4cb8548a412a789936d640a97b1aa0a; jvm 11.0.19+7
2023-09-26 11:37:42,031 INFO  [org.ecl.jet.ser.han.ContextHandler] (build-2) Started o.e.j.s.ServletContextHandler@17538702{/__admin,null,AVAILABLE}
2023-09-26 11:37:42,036 INFO  [org.ecl.jet.ser.han.Con.ROOT] (build-2) RequestHandlerClass from context returned com.github.tomakehurst.wiremock.http.StubRequestHandler. Normalized mapped under returned 'null'
2023-09-26 11:37:42,036 INFO  [org.ecl.jet.ser.han.ContextHandler] (build-2) Started o.e.j.s.ServletContextHandler@2b09608{/,null,AVAILABLE}
2023-09-26 11:37:42,047 INFO  [org.ecl.jet.ser.AbstractConnector] (build-2) Started NetworkTrafficServerConnector@5f63ed01{HTTP/1.1, (http/1.1, h2c)}{0.0.0.0:8089}
2023-09-26 11:37:42,047 INFO  [org.ecl.jet.ser.Server] (build-2) Started @1550ms
2023-09-26 11:37:42,170 INFO  [org.ecl.jet.ser.han.Con.__admin] (qtp1721857395-27) RequestHandlerClass from context returned com.github.tomakehurst.wiremock.http.AdminRequestHandler. Normalized mapped under returned 'null'
Executing "/Users/wjglerum/.sdkman/candidates/java/11.0.19-tem/bin/java -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/Users/wjglerum/Documents/Dev/quarkus-wiremock/integration-tests/target/quarkus.log -Dquarkus.log.file.enable=true -Dquarkus.wiremock.devservices.port=8089 -jar /Users/wjglerum/Documents/Dev/quarkus-wiremock/integration-tests/target/quarkus-app/quarkus-run.jar"
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2023-09-26 11:37:42,892 INFO  [io.quarkus] (main) quarkus-wiremock-integration-tests 0.0.3-SNAPSHOT on JVM (powered by Quarkus 2.16.5.Final) started in 0.409s. Listening on: http://0.0.0.0:8081
2023-09-26 11:37:42,900 INFO  [io.quarkus] (main) Profile prod activated. 
2023-09-26 11:37:42,900 INFO  [io.quarkus] (main) Installed features: [cdi, rest-client-reactive, resteasy-reactive, smallrye-context-propagation, vertx, wiremock]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.830 s -- in io.quarkiverse.wiremock.devservices.WireMockDevServiceResourceIT
```
Now the extension is correctly included in the jar

Reply:
For sure, IMHO this isn't the intended behavior. At least my first impression is why should this extension be part of the final jar? It's just an extension for the ``test`` and ``dev`` lifecycle. Of course, the code makes sure that noting happens at runtime but it seems a little bit odd to me to have this extension present in the ``prod`` build. That's why I've added ``provided`` ..

Reply:
It needs to be part of the prod build, as otherwise it's not part of the final jar that's started by the QuarkusIntegrationTest, see https://quarkus.io/guides/getting-started-testing#quarkus-integration-test

Reply:
I do get your point I think, normally a user of this extension wouldn't need to include this extension for the prod profile. But if you want to run it using the `QuarkusIntegrationTest` we kinda have to

Reply:
From my understanding it does exactly what it should do even in case the extension is not part of the final jar. Let's have look into your log example:

```
[INFO] Running io.quarkiverse.wiremock.devservices.WireMockDevServiceResourceIT
2023-09-26 11:37:41,725 DEBUG [io.qua.wir.dev.WireMockDevProcessor] (build-2) Quarkus launch mode [TEST]
2023-09-26 11:37:41,726 DEBUG [io.qua.wir.dev.WireMockDevProcessor] (build-2) Starting WireMock server with port [8089] and path [src/test/resources]
2023-09-26 11:37:41,875 INFO  [org.ecl.jet.uti.log] (build-2) Logging initialized @1378ms to org.eclipse.jetty.util.log.Slf4jLog
2023-09-26 11:37:42,023 INFO  [org.ecl.jet.ser.Server] (build-2) jetty-9.4.49.v20220914; built: 2022-09-14T01:07:36.601Z; git: 4231a3b2e4cb8548a412a789936d640a97b1aa0a; jvm 11.0.19+7
2023-09-26 11:37:42,031 INFO  [org.ecl.jet.ser.han.ContextHandler] (build-2) Started o.e.j.s.ServletContextHandler@17538702{/__admin,null,AVAILABLE}
2023-09-26 11:37:42,036 INFO  [org.ecl.jet.ser.han.Con.ROOT] (build-2) RequestHandlerClass from context returned com.github.tomakehurst.wiremock.http.StubRequestHandler. Normalized mapped under returned 'null'
2023-09-26 11:37:42,036 INFO  [org.ecl.jet.ser.han.ContextHandler] (build-2) Started o.e.j.s.ServletContextHandler@2b09608{/,null,AVAILABLE}
2023-09-26 11:37:42,047 INFO  [org.ecl.jet.ser.AbstractConnector] (build-2) Started NetworkTrafficServerConnector@5f63ed01{HTTP/1.1, (http/1.1, h2c)}{0.0.0.0:8089}
2023-09-26 11:37:42,047 INFO  [org.ecl.jet.ser.Server] (build-2) Started @1550ms
2023-09-26 11:37:42,170 INFO  [org.ecl.jet.ser.han.Con.__admin] (qtp1721857395-27) RequestHandlerClass from context returned com.github.tomakehurst.wiremock.http.AdminRequestHandler. Normalized mapped under returned 'null'
Executing "/Users/wjglerum/.sdkman/candidates/java/11.0.19-tem/bin/java -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/Users/wjglerum/Documents/Dev/quarkus-wiremock/integration-tests/target/quarkus.log -Dquarkus.log.file.enable=true -Dquarkus.wiremock.devservices.port=8089 -jar /Users/wjglerum/Documents/Dev/quarkus-wiremock/integration-tests/target/quarkus-app/quarkus-run.jar"
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2023-09-26 11:37:42,892 INFO  [io.quarkus] (main) quarkus-wiremock-integration-tests 0.0.3-SNAPSHOT on JVM (powered by Quarkus 2.16.5.Final) started in 0.409s. Listening on: http://0.0.0.0:8081
2023-09-26 11:37:42,900 INFO  [io.quarkus] (main) Profile prod activated. 
2023-09-26 11:37:42,900 INFO  [io.quarkus] (main) Installed features: [cdi, rest-client-reactive, resteasy-reactive, smallrye-context-propagation, vertx, wiremock]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.830 s -- in io.quarkiverse.wiremock.devservices.WireMockDevServiceResourceIT
```

The ``WireMockDevProcessor``  is being called in ``Quarkus launch mode [TEST]`` although the ``quarkus-wiremock-integration-tests`` app is started in ``Profile prod``. IMHO their is a difference between the ``launch mode`` and the ``profile`` the application has been started. And my ``integration-test`` already proves, that the wiremock-server is running for ITs, although the extension is not present in the final jar.

Reply:
I do see that indeed. Below is what is happening from my understanding:

The wiremock server is indeed running during the QuarkusIntegrationTest, because it's started during the build phase of the quarkus app. However the resulting Quarkus app that's packaged doesn't include the wiremock extension (as you can see in the logs). The test passes because wiremock is already started but not actually part of the running app.

This is what you can see from the thread id's in the logs. The Wiremock server is started from one of the `build-` threads.

It's an interesting case, because for most extensions the extension itself will be present on prod and will handle the properties and ignore/disable the devservices. However in this case we probably only want the wiremock extension enabled for dev/test and not in prod.

I was also browsing through some more extensions and integration tests but couldn't find any other examples of integrations that do this.

Reply:
We might want to ask someone from the Quarkus team to help us out here

Reply:
> I do see that indeed. Below is what is happening from my understanding:
> 
> The wiremock server is indeed running during the QuarkusIntegrationTest, because it's started during the build phase of the quarkus app. However the resulting Quarkus app that's packaged doesn't include the wiremock extension (as you can see in the logs). The test passes because wiremock is already started but not actually part of the running app.
> 
> This is what you can see from the thread id's in the logs. The Wiremock server is started from one of the `build-` threads.
> 

Exactly, it's the responsibility of a build step to start the ``wiremock-server``. Basically, that's the purpose of this extension. I don't see why the extension should be part of the application under test. The integration tests are also not part of the final app. Why should they? Nothing I'd like to have in my prod build ...

> It's an interesting case, because for most extensions the extension itself will be present on prod and will handle the properties and ignore/disable the devservices. However in this case we probably only want the wiremock extension enabled for dev/test and not in prod.

Even if we would enable the ``extension`` in ``prod``, what would it do? The whole functionality of that extension is done at ``build-time`` (let's ignore the ``wiremock-test`` dependency for the time being).
 
> I was also browsing through some more extensions and integration tests but couldn't find any other examples of integrations that do this.

Sure, @gastaldi wdyt?

Reply:
AFAIU this extension shiuld be used in `<scope>test</scope>` so it won't end up in the production JAR and it is available during dev mode

Reply:
@gastaldi Thanks for your response. So the ``quarkus.builder`` prefix does the trick to suppress the warning?

Reply:
> @gastaldi Thanks for your response. So the `quarkus.builder` prefix does the trick to suppress the warning?

In the `BuilderConfig` case, it maps `quarkus.builder.graph-output` and `quarkus.builder.log-conflict-cause` to the interface methods. You need a similar approach for `quarkus.wiremock`

Reply:
Yeah, but in your example this ``@ConfigMapping`` is part of the ``runtime`` module. We just decided that we won't have the runtime module present in ``prod`` mode. IMHO: We'll still face this warning ...

Actually I don't understand why ``Quarkus`` adds this parameter `` -Dquarkus.wiremock.devservices.port=8089`` when starting the application under test:
```
2023-09-26 11:37:42,170 INFO  [org.ecl.jet.ser.han.Con.__admin] (qtp1721857395-27) RequestHandlerClass from context returned com.github.tomakehurst.wiremock.http.AdminRequestHandler. Normalized mapped under returned 'null'
Executing "/Users/wjglerum/.sdkman/candidates/java/11.0.19-tem/bin/java -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/Users/wjglerum/Documents/Dev/quarkus-wiremock/integration-tests/target/quarkus.log -Dquarkus.log.file.enable=true -Dquarkus.wiremock.devservices.port=8089 -jar /Users/wjglerum/Documents/Dev/quarkus-wiremock/integration-tests/target/quarkus-app/quarkus-run.jar"
```

The warning would be gone if we could avoid that ``Quarkus`` would add this parameter to the application under test.

Reply:
I don't think there will ever exist a `prod` mode, as this extension is supposed to be used in `test` mode only? This parameter is added because it's part of the  `DevServicesResultBuildItem` produced in https://github.com/quarkiverse/quarkus-wiremock/blob/main/deployment/src/main/java/io/quarkiverse/wiremock/devservices/QuarkusWireMockDevProcessor.java#L107

Reply:
Yes, this extension is supposed to be used in ``test`` mode only. However, during ``integration tests`` we've passed this parameter ``-Dquarkus.wiremock.devservices.port=8089`` to the application under test. This was not by intention so we got this ``unrecognized property warning`` because the ``wiremock`` extension was not present anymore. But now you pointed us to the right direction so I guess we're good to go.

@gastaldi Thank you very much for your help!